### ISSUE TITLE
[1.11] JEI not working after Ore Dictionary nullability fixes

### ISSUE THREAD

**Issue by Choonster** [mezz/JustEnoughItems#626]
MinecraftForge/MinecraftForge@bf090cb4c2f76dc4bd6f58f7cbcb6b3487407dae fixed the `ItemStack` nullability issues with the Ore Dictionary.

This means that `IRecipe#getRecipeOutput` always returns a non-`null` value for ore recipes and `ShapelessOreRecipe#getInput` returns a `NonNullList` instead of an `ArrayList`.

The latter is causing a `NoSuchMethodError` at runtime from `ShapelessOreRecipeHandler#isRecipeValid` because the return type was included in the bytecode (even though JEI never explicitly uses it as an `ArrayList`).

```
[09:44:01] [Server thread/ERROR] [FML/] []: Caught exception from Just Enough Items (jei)
java.lang.NoSuchMethodError: net.minecraftforge.oredict.ShapelessOreRecipe.getInput()Ljava/util/ArrayList;
	at mezz.jei.plugins.vanilla.crafting.ShapelessOreRecipeHandler.isRecipeValid(ShapelessOreRecipeHandler.java:43) ~[jei_1.11-4.0.1.190.jar:?]
	at mezz.jei.plugins.vanilla.crafting.ShapelessOreRecipeHandler.isRecipeValid(ShapelessOreRecipeHandler.java:13) ~[jei_1.11-4.0.1.190.jar:?]
	at mezz.jei.RecipeRegistry.addRecipe(RecipeRegistry.java:206) ~[jei_1.11-4.0.1.190.jar:?]
	at mezz.jei.RecipeRegistry.addRecipe(RecipeRegistry.java:183) ~[jei_1.11-4.0.1.190.jar:?]
	at mezz.jei.RecipeRegistry.addRecipes(RecipeRegistry.java:160) ~[jei_1.11-4.0.1.190.jar:?]
	at mezz.jei.RecipeRegistry.<init>(RecipeRegistry.java:86) ~[jei_1.11-4.0.1.190.jar:?]
	at mezz.jei.util.ModRegistry.createRecipeRegistry(ModRegistry.java:195) ~[jei_1.11-4.0.1.190.jar:?]
	at mezz.jei.JeiStarter.start(JeiStarter.java:48) ~[jei_1.11-4.0.1.190.jar:?]
	at mezz.jei.ProxyCommonClient.loadComplete(ProxyCommonClient.java:111) ~[jei_1.11-4.0.1.190.jar:?]
	at mezz.jei.JustEnoughItems.loadComplete(JustEnoughItems.java:53) ~[jei_1.11-4.0.1.190.jar:?]
```

[Full Log](https://gist.github.com/Choonster/f73d815dc58a71bf45186be613425247)

**Comment by mezz** [comment]
Thanks for the heads up!
This is a good change.

**Comment by mezz** [comment]
I think the bytecode change should be reversed, there's no benefit to having it as a `NonNullList`.
In any case I'll build a new JEI to work with this.

**Comment by mezz** [comment]
Should be fixed in 4.0.1.191

**Comment by Choonster** [comment]
Thanks for the quick fix.
